tests/system_tests_http1_base.py

```diff
@@ -0,0 +1,1187 @@
+             ResponseMsg(201, reason="Created",
+                         headers={"Response-Header": "whatever",
+                                  "Content-length": "3"},
+                         body=b'ABC'),
+             ResponseValidator(status=201, expect_body=b'ABC')
+             ),
+
+            (RequestMsg("PUT", "/PUT/test_02",
+                        headers={"Put-Header-1": "Value 01",
+                                 "Content-length": "0",
+                                 "Content-Type": "text/plain;charset=utf-8"}),
+             ResponseMsg(201, reason="Created",
+                         headers={"Response-Header": "whatever"},
+                         body=b'No Content Length'),
+             ResponseValidator(status=201, expect_body=b'No Content Length')
+             ),
+        ]
+    }
+
+    def _do_request(self, client, tests):
+        for req, _, val in tests:
+            req.send_request(client)
+            rsp = client.getresponse()
+            try:
+                body = val.check_response(rsp)
+            except Exception as exc:
+                self.fail("request failed:  %s" % str(exc))
+
+            if req.method == "BODY":
+                self.assertEqual(b'', body)
+
+    def test_001_get(self):
+        client = HTTPConnection("127.0.0.1:%s" % self.http_listener11_port,
+                                timeout=TIMEOUT)
+        self._do_request(client, self.TESTS_11["GET"])
+        client.close()
+
+    def test_002_head(self):
+        client = HTTPConnection("127.0.0.1:%s" % self.http_listener11_port,
+                                timeout=TIMEOUT)
+        self._do_request(client, self.TESTS_11["HEAD"])
+        client.close()
+
+    def test_003_post(self):
+        client = HTTPConnection("127.0.0.1:%s" % self.http_listener11_port,
+                                timeout=TIMEOUT)
+        self._do_request(client, self.TESTS_11["POST"])
+        client.close()
+
+    def test_004_put(self):
+        client = HTTPConnection("127.0.0.1:%s" % self.http_listener11_port,
+                                timeout=TIMEOUT)
+        self._do_request(client, self.TESTS_11["PUT"])
+        client.close()
+
+    def test_006_head_10(self):
+        client = HTTPConnection("127.0.0.1:%s" % self.http_listener10_port,
+                                timeout=TIMEOUT)
+        self._do_request(client, self.TESTS_10["HEAD"])
+        client.close()
+
+    def test_007_post_10(self):
+        client = HTTPConnection("127.0.0.1:%s" % self.http_listener10_port,
+                                timeout=TIMEOUT)
+        self._do_request(client, self.TESTS_10["POST"])
+        client.close()
+
+    def test_008_put_10(self):
+        client = HTTPConnection("127.0.0.1:%s" % self.http_listener10_port,
+                                timeout=TIMEOUT)
+        self._do_request(client, self.TESTS_10["PUT"])
+        client.close()
+
+
+class Http1OneRouterTestBase(TestCase):
```

Comment:
Why is this broken up into two classes: Http1OneRouterTestBase and CommonHttp1AdaptorOneRouterTest?   Since they must be inherited from together why not just combine them into a single class?

Reply:
The CommonHttp1AdaptorOneRouterTest contains only the tests, it does not have any setup information. In the future, this separation enables the the battery of tests in CommonHttp1AdaptorOneRouterTest to be run against a totally different setup